lib/document/validateDocumentFactory.js

```diff
@@ -47,6 +48,14 @@ module.exports = function validateDocumentFactory(
 
     const result = new ValidationResult();
 
+    if (!Object.prototype.hasOwnProperty.call(rawDocument, '$id')) {
```

Comment:
We already check it with JSON Schema

Reply:
Then why do we check it for type?

Reply:
Because we use $type before validation in order to get document JSON Schema